<?xml version="1.0"?>
<launch>
    <node pkg="ring_detector" type="find_plane" name="find_plane" output="screen">
        <remap from="input" to="/camera/depth_registered/points"/>
        <remap from="/ring_detector/find_plane" to="plane"/>
    </node>
    	<node pkg="ring_detector" type="cloud_voxelgrid" name="voxelgrid" output="screen">
		<remap from="input" to="/plane"/>
		<remap from="output" to="/ring_detector/voxelgrid"/>
	</node>
	<node pkg="ring_detector" type="contraster" name="contraster" output="screen">
	        <remap from="input" to="/ring_detector/voxelgrid"/>
	</node>     
	<node pkg="ring_detector" type="find_circ" name="find_circB" args="blue" output="screen">
		<remap from="input" to="pclB"/>
		<remap from="marker" to="/ring_detector/circs"/>
		<remap from="pose" to="/ring_detector/circ_pose"/>
	</node>
	<node pkg="ring_detector" type="find_circ" name="find_circG"  args="green" output="screen">
		<remap from="input" to="pclG"/>
		<remap from="marker" to="/ring_detector/circs"/>
		<remap from="pose" to="/ring_detector/circ_pose"/>
	</node>
	<node pkg="ring_detector" type="find_circ" name="find_circR" args="red"  output="screen">
		<remap from="input" to="pclR"/>
		<remap from="marker" to="/ring_detector/circs"/>
		<remap from="pose" to="/ring_detector/circ_pose"/>
	</node> 
	<node pkg="ring_detector" type="find_circ" name="find_circK"  args="black" output="screen">
		<remap from="input" to="pclK"/>
		<remap from="marker" to="/ring_detector/circs"/>
		<remap from="pose" to="/ring_detector/circ_pose"/>
	</node>    
	<node pkg="ring_detector" type="find_cyl" name="find_cylR" args="red"  output="screen">
	        <remap from="input" to="pclR"/>
	        <remap from="cylinder" to="/ring_detector/find_cyl"/>
	</node>	    
	<node pkg="ring_detector" type="find_cyl" name="find_cylG" args="green"  output="screen">
	        <remap from="input" to="pclG"/>
	        <remap from="cylinder" to="/ring_detector/find_cyl"/>
	</node>	    
	<node pkg="ring_detector" type="find_cyl" name="find_cylB" args="blue"  output="screen">
	        <remap from="input" to="pclB"/>
	        <remap from="cylinder" to="/ring_detector/find_cyl"/>
	</node>	    
	<node pkg="ring_detector" type="find_cyl" name="find_cylY" args="yellow"  output="screen">
	        <remap from="input" to="pclY"/>
	        <remap from="cylinder" to="/ring_detector/find_cyl"/>
	</node>	
	<node pkg="ring_detector" type="seperate" name="seperate" output="screen">
	        <remap from="input" to="contrastedPcl"/>
	</node>


</launch>
